### PR TITLE
TST: stats: fix xslow test failures

### LINKED ISSUE #18680: stats:  XSLOW test failures in TestFit
I'm getting two failures in the stats `TestFit` test class when I run the tests with `SCIPY_XSLOW=1`:

```
$ SCIPY_XSLOW=1 python3 dev.py test -m full -j 10 -t scipy.stats.tests.test_fit::TestFit
💻  ninja -C /home/warren/repos/git/forks/scipy/build -j12
ninja: Entering directory `/home/warren/repos/git/forks/scipy/build'
[4/4] Generating scipy/generate-version with a custom command
Build OK
💻  meson install -C build --only-changed
Installing, see meson-install.log...
Installation OK
SciPy from development installed path at: /home/warren/repos/git/forks/scipy/build-install/lib/python3.10/site-packages
Running tests for scipy version:1.12.0.dev0+1169.9bc74cd, installed at:/home/warren/repos/git/forks/scipy/build-install/lib/python3.10/site-packages/scipy
=========================================== test session starts ===========================================
platform linux -- Python 3.10.8, pytest-7.1.3, pluggy-1.0.0
rootdir: /home/warren/repos/git/forks/scipy, configfile: pytest.ini
plugins: forked-1.4.0, timeout-2.1.0, hypothesis-6.42.1, xdist-2.5.0, cov-4.0.0
gw0 [266] / gw1 [266] / gw2 [266] / gw3 [266] / gw4 [266] / gw5 [266] / gw6 [266] / gw7 ok / gw8 ok / gw9 ogw0 [266] / gw1 [266] / gw2 [266] / gw3 [266] / gw4 [266] / gw5 [266] / gw6 [266] / gw7 [266] / gw8 ok / gwgw0 [266] / gw1 [266] / gw2 [266] / gw3 [266] / gw4 [266] / gw5 [266] / gw6 [266] / gw7 [266] / gw8 [266] /gw0 [266] / gw1 [266] / gw2 [266] / gw3 [266] / gw4 [266] / gw5 [266] / gw6 [266] / gw7 [266] / gw8 [266] / gw9 [266]
..............s..................s..........s................s.........s................ss......... [ 37%]
.ss............s...........................s.................ss.........s....s..s..........s....... [ 74%]
.....ss...........s.......................F.F.......................                                [100%]
================================================ FAILURES =================================================
___________________________________________ TestFit.test_guess ____________________________________________
[gw7] linux -- Python 3.10.8 /home/warren/py3.10.8/bin/python3
scipy/stats/tests/test_fit.py:698: in test_guess
    assert not np.allclose(res.params, params, **self.tols)
E   AssertionError: assert not True
E    +  where True = <function allclose at 0x7f621ed5feb0>(FitParams(M=20.0, n=7.0, r=12.0, loc=0.0), (20, 7, 12, 0), **{'atol': 0.05, 'rtol': 0.01})
E    +    where <function allclose at 0x7f621ed5feb0> = np.allclose
E    +    and   FitParams(M=20.0, n=7.0, r=12.0, loc=0.0) =   params: FitParams(M=20.0, n=7.0, r=12.0, loc=0.0)\n success: True\n message: 'Optimization terminated successfully.'.params
E    +    and   {'atol': 0.05, 'rtol': 0.01} = <scipy.stats.tests.test_fit.TestFit object at 0x7f61154cfdc0>.tols
        N          = 2000
        bounds     = [(2, 200), (0.7, 70), (1.2, 120), (0, 10)]
        data       = array([6, 4, 7, ..., 5, 7, 7])
        dist       = <scipy.stats._discrete_distns.nhypergeom_gen object at 0x7f6115a92860>
        params     = (20, 7, 12, 0)
        res        =   params: FitParams(M=20.0, n=7.0, r=12.0, loc=0.0)
 success: True
 message: 'Optimization terminated successfully.'
        rng        = Generator(PCG64) at 0x7F611546FCA0
        self       = <scipy.stats.tests.test_fit.TestFit object at 0x7f61154cfdc0>
_________________________________ TestFit.test_basic_fit_mse[tukeylambda] _________________________________
[gw3] linux -- Python 3.10.8 /home/warren/py3.10.8/bin/python3
scipy/stats/tests/test_fit.py:524: in test_basic_fit_mse
    self.basic_fit_test(dist_name, "mse")
        dist_name  = 'tukeylambda'
        self       = <scipy.stats.tests.test_fit.TestFit object at 0x7fa6eb0a6bf0>
scipy/stats/tests/test_fit.py:515: in basic_fit_test
    assert_nlff_less_or_close(dist, data, res.params, ref, **self.tols,
        N          = 5000
        bounds     = array([[3.13214779e-01, 3.13214779e+01],
       [0.00000000e+00, 1.00000000e+01],
       [1.00000000e-16, 1.00000000e+01]])
        data       = array([4.9621036 , 4.60024271, 4.63491397, ..., 4.79133554, 4.82754466,
       4.95822271])
        dist       = <scipy.stats._continuous_distns.tukeylambda_gen object at 0x7fa6eb976bf0>
        dist_data  = {'alpha': (3.570477051665046,), 'anglit': (), 'arcsine': (), 'argus': (1.0,), ...}
        dist_name  = 'tukeylambda'
        loc        = 4.790547883456026
        method     = 'mse'
        nlff_name  = '_penalized_nlpsf'
        nlff_names = {'mle': 'nnlf', 'mse': '_penalized_nlpsf'}
        ref        = [3.1321477856738267, 4.790547883456026, 0.9992905124308905]
        res        =   params: FitParams(lam=5.138048555592726, loc=4.7788093214154985, scale=1.6997288425156312)
 success: True
 message: 'Optimization terminated successfully.'
        rng        = Generator(PCG64) at 0x7FA6EB090900
        scale      = 0.9992905124308905
        self       = <scipy.stats.tests.test_fit.TestFit object at 0x7fa6eb0a6bf0>
        shapes     = array([3.13214779])
        sup        = <numpy.testing._private.utils.suppress_warnings object at 0x7fa6eb0a7ca0>
scipy/stats/tests/test_fit.py:351: in assert_nlff_less_or_close
    np.testing.assert_allclose(nlff1, nlff0, rtol=rtol, atol=atol)
        atol       = 0.05
        data       = array([4.9621036 , 4.60024271, 4.63491397, ..., 4.79133554, 4.82754466,
       4.95822271])
        dist       = <scipy.stats._continuous_distns.tukeylambda_gen object at 0x7fa6eb976bf0>
        nlff       = <bound method rv_generic._penalized_nlpsf of <scipy.stats._continuous_distns.tukeylambda_gen object at 0x7fa6eb976bf0>>
        nlff0      = 45522.43302746734
        nlff1      = 46064.8192201478
        nlff_name  = '_penalized_nlpsf'
        params0    = [3.1321477856738267, 4.790547883456026, 0.9992905124308905]
        params1    = FitParams(lam=5.138048555592726, loc=4.7788093214154985, scale=1.6997288425156312)
        rtol       = 0.01
/home/warren/py3.10.8/lib/python3.10/contextlib.py:79: in inner
    return func(*args, **kwds)
E   AssertionError: 
E   Not equal to tolerance rtol=0.01, atol=0.05
E   
E   Mismatched elements: 1 / 1 (100%)
E   Max absolute difference: 542.38619268
E   Max relative difference: 0.0119147
E    x: array(46064.81922)
E    y: array(45522.433027)
        args       = (<function assert_allclose.<locals>.compare at 0x7fa6eaf72dd0>, array(46064.81922015), array(45522.43302747))
        func       = <function assert_array_compare at 0x7fa7b3bfab00>
        kwds       = {'equal_nan': True, 'err_msg': '', 'header': 'Not equal to tolerance rtol=0.01, atol=0.05', 'verbose': True}
        self       = <contextlib._GeneratorContextManager object at 0x7fa7b3de2cb0>
========================================= short test summary info =========================================
FAILED scipy/stats/tests/test_fit.py::TestFit::test_guess - AssertionError: assert not True
FAILED scipy/stats/tests/test_fit.py::TestFit::test_basic_fit_mse[tukeylambda] - AssertionError: 
=============================== 2 failed, 244 passed, 20 skipped in 41.49s ================================
```

System information:
```
In [1]: import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info);
   ...: scipy.show_config()
1.12.0.dev0+1169.9bc74cd 1.24.3 sys.version_info(major=3, minor=10, micro=8, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: /usr/include/x86_64-linux-gnu/openblas-pthread/
    lib directory: /usr/lib/x86_64-linux-gnu/openblas-pthread/
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER=1 NO_CBLAS=
      NO_LAPACK= NO_LAPACKE=1 NO_AFFINITY=1 USE_OPENMP=0 generic MAX_THREADS=64
    pc file directory: /usr/lib/x86_64-linux-gnu/pkgconfig
    version: 0.3.20
  lapack:
    detection method: pkgconfig
    found: true
    include directory: /usr/include/x86_64-linux-gnu/openblas-pthread/
    lib directory: /usr/lib/x86_64-linux-gnu/openblas-pthread/
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER=1 NO_CBLAS=
      NO_LAPACK= NO_LAPACKE=1 NO_AFFINITY=1 USE_OPENMP=0 generic MAX_THREADS=64
    pc file directory: /usr/lib/x86_64-linux-gnu/pkgconfig
    version: 0.3.20
  pybind11:
    detection method: config-tool
    include directory: unknown
    name: pybind11
    version: 2.10.4
Compilers:
  c:
    commands: cc
    linker: ld.bfd
    name: gcc
    version: 11.3.0
  c++:
    commands: c++
    linker: ld.bfd
    name: gcc
    version: 11.3.0
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 0.29.35
  fortran:
    commands: gfortran
    linker: ld.bfd
    name: gcc
    version: 11.3.0
  pythran:
    include directory: ../../../../../py3.10.8/lib/python3.10/site-packages/pythran
    version: 0.12.1
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
Python Information:
  path: /home/warren/py3.10.8/bin/python3
  version: '3.10'

```

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
Closes gh-18680

#### What does this implement/fix?
gh-18680 reported failures in xslow tests. This fixes them.

#### Additional Information
These probably started failing due to gh-18496 (effectively) changing the random number stream.

### CODE REVIEW

File: scipy/stats/tests/test_fit.py [hunk 1]
```diff
@@ -269,7 +269,7 @@ def cases_test_fit_mse():
                       'kstwo',  # very slow (~25 min) but passes
                       'geninvgauss',  # quite slow (~4 minutes) but passes
                       'gausshyper', 'genhyperbolic',  # integration warnings
-                      'argus',  # close, but doesn't meet tolerance
+                      'argus', 'tukeylambda',  # close, but doesn't meet tolerance
```
Comment (mdhaber) [inline scipy/stats/tests/test_fit.py:272]:
```suggestion
                      'tukeylambda',  # close, but doesn't meet tolerance
```
`argus` seems to pass now.

**Review by tupui — APPROVED** [review]
Thanks Matt, LGTM (CI is just red because your branch is behind `main`.)

Do you want to get your suggestion in, or should we merge now?

**Review by tupui — APPROVED** [review]
Ok it's all good locally on my side. Please double check I understood you correctly and go ahead if I did.

**Comment by mdhaber** [comment]
Yeah, we just need to run CI because taking `argus` out of the xfail list means that test will run on all platforms. I merged main, so if if it comes back green, this is good to go.

**Comment by mdhaber** [comment]
Oops no, you took `tukeylambda` out of the MLE xslow list. It needed to be removed from the MSE `xslow` list because it is in the MSE `skip` list. But argus passed this time, so we don't need to run CI again.

File: scipy/stats/tests/test_fit.py [hunk 2]
```diff
@@ -269,7 +269,7 @@ def cases_test_fit_mse():
                       'kstwo',  # very slow (~25 min) but passes
                       'geninvgauss',  # quite slow (~4 minutes) but passes
                       'gausshyper', 'genhyperbolic',  # integration warnings
-                      'argus',  # close, but doesn't meet tolerance
```
Comment (mdhaber) [inline scipy/stats/tests/test_fit.py:272]: `argus` MSE now meets tolerance, as evidenced by CI passing

File: scipy/stats/tests/test_fit.py [hunk 3]
```diff
@@ -269,7 +269,7 @@ def cases_test_fit_mse():
...
+                      'tukeylambda',  # close, but doesn't meet tolerance
```
Comment (mdhaber) [inline scipy/stats/tests/test_fit.py:272]: `tukeylambda` MSE does not (see report in gh-18680)

File: scipy/stats/tests/test_fit.py [hunk 4]
```diff
@@ -300,7 +300,7 @@ def cases_test_fit_mse():
                        'nct', 'ncx2',
                        'pearson3', 'powerlaw', 'powerlognorm',
                        'rdist', 'reciprocal', 'rel_breitwigner', 'rice',
-                       'trapezoid', 'truncnorm', 'tukeylambda',
+                       'trapezoid', 'truncnorm',
```
Comment (mdhaber) [inline scipy/stats/tests/test_fit.py:303]: `tukeylambda` MSE need not be marked `xslow` anymore, since it's skipped.

**Review by mdhaber — COMMENTED** [review]
@tupui OK, now I think this is done. Please feel free to merge.

**Review by tupui — APPROVED** [review]
Ok good that I asked to double check then 😅

**Pull request merged**